src/DssDirectDepositHub.sol

```diff
@@ -215,7 +214,7 @@ contract DssDirectDepositHub {
         // This module will have an unintended behaviour if rate is changed to some other value.
 
         address end;
-        uint256 gemBalance = pool.gemBalanceOf();
+        uint256 gemBalance = pool.maxRedeem();
```

Comment:
This needs to be reworked to use `convertToAssets(balanceOf())` to get the total position size in DAI, then the amount to withdraw is constrained by `maxWithdraw()` to get `availableLiquidity`.

Would also stick with language `asset` for the language instead of `gem`. IE `gemBalance` -> `assetBalance`.

Reply:
I think this should be shares. Gem at this point was the adai balance.  Let me know if that makes sense to you